From 9711533bbab9260399871e893028f46ccd8961ed Mon Sep 17 00:00:00 2001
From: Elvis Lee <kwangwoong.lee@lge.com>
Date: Wed, 11 Mar 2015 17:16:13 +0900
Subject: [PATCH] Revert "Fix rendering of fonts matched based on stretch"

:Release Notes:
Revert fix for rendering of fonts matched based on stretch.

:Detailed Notes:
This reverts commit a128347c9b51fc9fa629662a14b77e1489537276.
https://codereview.qt-project.org/#/c/92219/

This is workaround to avoid taking the stretched font flavour and
applying again the stretch factor.
Our final goal is to fix the fonts themselve.
Then, we can restore this patch agian.

:Issues Addressed:
[DRD-3715] Font sizes seem to be larger in Qt 5.4 build
[DRD-5249] Review all fonts after migrating to Qt5.4

Change-Id: Ic8a4ddcbe29e13f1008700ca5859982f34bd0f3c
Upstream-Status: do not apply (local workaround)
---
 src/gui/text/qfontdatabase.cpp | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/src/gui/text/qfontdatabase.cpp b/src/gui/text/qfontdatabase.cpp
index f147851..6e48de1 100644
--- a/src/gui/text/qfontdatabase.cpp
+++ b/src/gui/text/qfontdatabase.cpp
@@ -953,15 +953,6 @@ QFontEngine *loadSingleEngine(int script,
                 return engine;
             }
         }
-
-        // If the font data's native stretch matches the requested stretch we need to set stretch to 100
-        // to avoid the fontengine synthesizing stretch. If they didn't match exactly we need to calculate
-        // the new stretch factor. This only done if not matched by styleName.
-        if (style->key.stretch != 0 && request.stretch != 0
-            && (request.styleName.isEmpty() || request.styleName != style->styleName)) {
-            def.stretch = (request.stretch * 100 + 50) / style->key.stretch;
-        }
-
         engine = pfdb->fontEngine(def, size->handle);
         if (engine) {
             // Also check for OpenType tables when using complex scripts
